### PR TITLE
Add HttpPostedFile*.

### LINKED ISSUE #141: Add HttpRequestBase.Files and supporting types
In PR https://github.com/dotnet/systemweb-adapters/pull/103 the types to handle file upload were added: Files property on HttpRequest and HttpPostedFile class.

These should also be created in the *Base variants: Files property in HttpRequestBase and a HttpPostedFileBase class.

### PR DESCRIPTION
**Pull request by johnLwith**
Add missing property of File in HttpRequest.

Fixes #141

### CODE REVIEW

File: src/Microsoft.AspNetCore.SystemWebAdapters/HttpFileCollection.cs [hunk 1]
```diff
@@ -43,10 +43,14 @@ public override IEnumerator GetEnumerator()
 
     public HttpPostedFile? Get(string name) => FormFiles.GetFile(name) is { } file ? new(file) : null;
 
+    public HttpPostedFile? Get(int index) => FormFiles[index] is { } file ? new(file) : null;
+
     public IList<HttpPostedFile> GetMultiple(string name) => FormFiles.GetFiles(name) is { Count: > 0 } files ? new ReadOnlyPostedFileCollection(files) : Array.Empty<HttpPostedFile>();
 
     public HttpPostedFile? this[string name] => Get(name);
 
+    public HttpPostedFile? this[int index] => Get(index);
```
Comment (twsouthwick) [inline src/Microsoft.AspNetCore.SystemWebAdapters/HttpFileCollection.cs:52]: @Tratcher I think this is the first of the namevaluecollection types we can actually fully conform - I hadn't realized IFormFileCollection had a `this[int index]` parameter

**Review by twsouthwick — APPROVED** [review]
LGTM

File: src/Microsoft.AspNetCore.SystemWebAdapters/HttpFileCollection.cs (hunk 1 above)
Comment (Tratcher) [reply src/Microsoft.AspNetCore.SystemWebAdapters/HttpFileCollection.cs:52]: Ha, yeah, it's a full IReadOnlyList.

File: src/Microsoft.AspNetCore.SystemWebAdapters/HttpPostedFileWrapper.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+
+namespace System.Web;
+
+public class HttpPostedFileWrapper : HttpPostedFileBase
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.SystemWebAdapters/HttpPostedFileWrapper.cs:8]: Are we making all of the wrapper types public? Why?

**Review by Tratcher — APPROVED** [review]

File: src/Microsoft.AspNetCore.SystemWebAdapters/HttpPostedFileWrapper.cs (hunk 2 above)
Comment (twsouthwick) [reply src/Microsoft.AspNetCore.SystemWebAdapters/HttpPostedFileWrapper.cs:8]: This is a public API from .NET Framework: https://learn.microsoft.com/en-us/dotnet/api/system.web.httppostedfilewrapper?view=netframework-4.8.1

It's a convention used for a number of types. Most common is `HttpContext` which has no virtual methods. `HttpContextBase` was an abstraction for it. `HttpContextWrapper` wrapped `HttpContext` to `HttpContextBase`.

**Comment by twsouthwick** [comment]
@johnLwith this is approved, but we've updated to start using .NET 8 SDK, and it enabled new warnings. Can you address them and then we'll merge it in

File: src/Microsoft.AspNetCore.SystemWebAdapters/HttpPostedFileBase.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+
+namespace System.Web;
+
+public abstract class HttpPostedFileBase
+{
+    public virtual string FileName => throw new NotImplementedException();
```
Comment (twsouthwick) [inline src/Microsoft.AspNetCore.SystemWebAdapters/HttpPostedFileBase.cs:10]: @johnLwith for these kind of warnings/errors, suppress with an attribute, and use the justification of `Constants.ApiFromAspNet`

**Comment by twsouthwick** [comment]
@johnLwith will you be able to address the build error?

**Comment by twsouthwick** [comment]
@johnLwith I went ahead and pushed a fix to your branch with the missing attributes

**Pull request merged**